Source/monster.cpp

```diff
@@ -5603,15 +5602,17 @@ void TalktoMonster(int i)
 	pnum = Monst->_menemy;
 	Monst->_mmode = MM_TALK;
 	if (Monst->_mAi == AI_SNOTSPIL || Monst->_mAi == AI_LACHDAN) {
-		if (QuestStatus(Q_LTBANNER) && quests[Q_LTBANNER]._qvar1 == 2 && PlrHasItem(pnum, IDI_BANNER, &itm)) {
-			RemoveInvItem(pnum, itm);
+		itm = plr[pnum].GetItemIndex(IDI_BANNER);
+		if (QuestStatus(Q_LTBANNER) && quests[Q_LTBANNER]._qvar1 == 2 && itm != -1) {
+			RemoveInvItem(&plr[pnum], itm);
 			quests[Q_LTBANNER]._qactive = QUEST_DONE;
 			Monst->mtalkmsg = TEXT_BANNER12;
 			Monst->_mgoal = MGOAL_INQUIRING;
 		}
 		if (QuestStatus(Q_VEIL) && Monst->mtalkmsg >= TEXT_VEIL9) {
-			if (PlrHasItem(pnum, IDI_GLDNELIX, &itm)) {
-				RemoveInvItem(pnum, itm);
+			itm = plr[pnum].GetItemIndex(IDI_GLDNELIX);

```

Comment:
This is very similar to the logic above, however we are doing it differently.

Would suggest doing the same checks the same way for consistency and readability (i.e. combine the `if`s and use the item as part of the condition).